x-pack/plugins/cases/public/components/templates/schema.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fieldValidators } from '@kbn/es-ui-shared-plugin/static/forms/helpers';
+import type { FormSchema } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { VALIDATION_TYPES } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import {
+  MAX_DESCRIPTION_LENGTH,
+  MAX_LENGTH_PER_TAG,
+  MAX_TAGS_PER_CASE,
+  MAX_TAGS_PER_TEMPLATE,
+  MAX_TEMPLATE_TAG_LENGTH,
+  MAX_TITLE_LENGTH,
+  MAX_TEMPLATE_NAME_LENGTH,
+  MAX_TEMPLATE_DESCRIPTION_LENGTH,
+} from '../../../common/constants';
+import { OptionalFieldLabel } from '../create/optional_field_label';
+import { SEVERITY_TITLE } from '../severity/translations';
+import * as i18n from './translations';
+import type { TemplateFormProps } from './types';
+import { validateEmptyTags, validateMaxLength, validateMaxTagsLength } from './utils';
+
+const { emptyField, maxLengthField } = fieldValidators;
+
+export const schema: FormSchema<TemplateFormProps> = {
```

Comment:
A portion of this schema seems identical with the schema defined here `x-pack/plugins/cases/public/components/create/schema.tsx`. Wdyt about having one base schema common to both? Then, in this file, we can iterate the common schema and add the optional label to the fields we want.

Reply:
Created schema for case_form_fields and used it in template schema to append optional label. [here](https://github.com/elastic/kibana/pull/184104/commits/e4aa6ad4eba559285810d6b57c630cb7d4812cf2#diff-80c709bae1c4a6584dc0c54ff67a4545847a65860e8323fcb649676301c79660)